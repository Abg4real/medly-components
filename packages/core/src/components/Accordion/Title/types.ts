import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp{
    color?: string;
    isActive?: boolean;
    onToggle?: () => void;
    bgColor?: string;

}


