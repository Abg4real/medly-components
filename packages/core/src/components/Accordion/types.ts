import { Props as TitleProps } from './Title';

export interface Props {
    Title?: React.SFC<TitleProps>;
    Content?: React.SFC;
}