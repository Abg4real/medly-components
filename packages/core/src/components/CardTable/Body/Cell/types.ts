import { ColumnConfig } from '../../types';

export interface Props {
    data: any;
    column: ColumnConfig;
    rowId: any;
    isRowClickDisabled?: boolean;
}

export interface StyledProps {
    align: 'left' | 'center' | 'right';
    wrapText: false | true;
}
