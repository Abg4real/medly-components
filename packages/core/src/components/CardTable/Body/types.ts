import { ColumnConfig, Data } from '../types';

export interface Props {
    data: Data[];
    columns: ColumnConfig[];
    onRowClick?: (rowData: object) => void;
    rowClickDisableKey?: string;
    uniqueKeyName?: string;
    withWhiteBackground?: boolean;
}
