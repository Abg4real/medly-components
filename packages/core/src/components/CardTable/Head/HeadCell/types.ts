import { WithThemeProp } from '@medly-components/utils/src';
import { ColumnConfig, SortOrder } from '../../types';

export interface Props extends WithThemeProp {
    column: ColumnConfig;
    sortField?: string;
    withWhiteBackground: boolean;
    onSortChange?: (field: string, order: SortOrder) => void;
}

export interface StyledProps {
    isActive: boolean;
    withWhiteBackground: boolean;
    flex: number;
    align: 'left' | 'center' | 'right';
}
