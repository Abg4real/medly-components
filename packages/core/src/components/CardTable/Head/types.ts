import { ColumnConfig, SortOrder } from '../types';

export interface Props {
    withWhiteBackground: boolean;
    columns: ColumnConfig[];
    onSort?: (field: string, order: SortOrder) => void;
}
