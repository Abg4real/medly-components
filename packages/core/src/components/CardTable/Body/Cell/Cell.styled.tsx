import { styled } from '@medly-components/utils';
import { StyledProps } from './types';

export const Cell = styled('td')<StyledProps>`
    flex: ${({ flex }) => flex};
    padding: 0;
    padding: 1.5rem 2.4rem;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: ${({ align }) => (align === 'left' ? 'flex-start' : align === 'right' ? 'flex-end' : 'center')};
`;
Cell.defaultProps = {
    flex: 1,
    align: 'left'
};
