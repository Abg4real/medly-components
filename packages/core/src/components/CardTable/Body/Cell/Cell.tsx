import { WithStyle } from '@medly-components/utils';
import React, { useMemo } from 'react';
import Text from '../../../Text';
import * as Style from './Cell.styled';
import { Props } from './types';

export const Cell: React.SFC<Props> & WithStyle = React.memo(props => {
    const { data, rowId, column, isRowClickDisabled } = props,
        { component: Component, align } = column;

    const FormattedCell = useMemo(
        () => (Component ? <Component {...{ data, rowId, disabled: isRowClickDisabled }} /> : <Text>{data}</Text>),
        [data, rowId, isRowClickDisabled, Component]
    );

    return <Style.Cell align={align}>{FormattedCell}</Style.Cell>;
});
Cell.displayName = 'Cell';
