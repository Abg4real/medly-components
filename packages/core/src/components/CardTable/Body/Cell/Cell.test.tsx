import { render } from '@testing-library/react';
import React from 'react';
import Text from '../../../Text';
import { ColumnConfig } from '../../types';
import { Cell } from './Cell';

const renderer = (columnConfig: ColumnConfig) =>
    render(
        <table>
            <tbody>
                <tr>
                    <Cell column={columnConfig} data="text" rowId="1" />
                </tr>
            </tbody>
        </table>
    );

describe('CardTable Cell', () => {
    it('should render cell with text properly', () => {
        const { container } = renderer({
            title: 'Column 1',
            field: 'column1'
        });
        expect(container).toMatchSnapshot();
    });

    it('should render cell with component properly', () => {
        const config: ColumnConfig = {
                title: 'Column 1',
                field: 'column1',
                component: Text
            },
            { container } = renderer(config);
        expect(container).toMatchSnapshot();
    });

    test.each(['left', 'center', 'right'])('should be able to %p align ', (align: 'left' | 'center' | 'right') => {
        const { container } = renderer({
            align,
            title: 'Column 1',
            field: 'column1'
        });
        expect(container).toMatchSnapshot();
    });
});
