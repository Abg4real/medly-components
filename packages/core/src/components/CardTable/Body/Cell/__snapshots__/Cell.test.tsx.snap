// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTable Cell should be able to "center" align  1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 1.5rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <table>
    <tbody>
      <tr>
        <td
          class="c0"
        >
          <span
            class="c1"
          >
            text
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CardTable Cell should be able to "left" align  1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 1.5rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div>
  <table>
    <tbody>
      <tr>
        <td
          class="c0"
        >
          <span
            class="c1"
          >
            text
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CardTable Cell should be able to "right" align  1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 1.5rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div>
  <table>
    <tbody>
      <tr>
        <td
          class="c0"
        >
          <span
            class="c1"
          >
            text
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CardTable Cell should render cell with component properly 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 1.5rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div>
  <table>
    <tbody>
      <tr>
        <td
          class="c0"
        >
          <span
            class="c1"
            data="text"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CardTable Cell should render cell with text properly 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 1.5rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div>
  <table>
    <tbody>
      <tr>
        <td
          class="c0"
        >
          <span
            class="c1"
          >
            text
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
